src/js/models/menubar.js

```diff
@@ -253,6 +253,16 @@ define(function (require, exports, module) {
                         return layer !== ancestor && document.layers.selected.contains(ancestor);
                     });
                 })),
+            "no-nesting-except-artboards":
+                (document !== null) &&
+                !document.unsupported &&
+                (document.layers !== null) &&
+                !(document.layers.selected.some(function (layer) {
+                    return document.layers.ancestors(layer).some(function (ancestor) {
```

Comment:
mild suggestion of putting this into `layerstructure` as a cached property ... since it seems slightly intensive.